### PR TITLE
Added three Data Structure and Algorithms course in Free Courses in English

### PR DESCRIPTION
**Pull request by moazamdev**
## What does this PR do?
This PR will add three Data Structure and Algorithms courses in Free Courses in English under Algorithms & Data Structures section

### Why is this valuable (or not)?
Best quality courses provided by top YouTubers 

### How do we know it's really free?
Courses are on YouTube

### For book lists, is it a book? For course lists, is it a course? etc.
These are courses for courses list 

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
For youtube, we prefer "playlist" links and the name of the producer/creator. https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md#platforms-and-access-notes
no shortened links

**Comment by moazamdev** [comment]
Hi @eshellman, I have added video length for the Data Structures Course by Simplilearn, Also I have mentioned the creator and the source of the course.

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -111,15 +111,18 @@
 * [Advanced Data Structures](http://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-851-advanced-data-structures-spring-2012/) - Erik Demaine
 * [Algorithm Design and Implementation](https://www.youtube.com/playlist?list=PL6EF0274BD849A7D5)
 * [Algorithms](https://www.youtube.com/playlist?list=PLDN4rrl48XKpZkf03iYFl-O29szjTrs_O) - Abdul Bari
+* [Algorithms and Data Structures Tutorial - Full Course for Beginners](https://youtu.be/8hly31xKli0) - freeCodeCamp (YouTube)
```
Comment (davorpa) [inline courses/free-courses-en.md:114]: Expand YT shorten urls

```suggestion
* [Algorithms and Data Structures Tutorial - Full Course for Beginners](https://www.youtube.com/watch?v=8hly31xKli0) - Pasan Premaratne, Jay McGavren (freeCodeCamp)
```

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -111,15 +111,18 @@
...
 * [Complete C++ Placement DSA Course](https://www.youtube.com/playlist?list=PLDzeHZWIZsTryvtXdMr6rPh4IDexB5NIA) - CodeHelp - by Babbar
 * [Data Structures](https://www.youtube.com/playlist?list=PL2_aWCzGMAwI3W_JlcBbtYTwiQSsOTa6P) - mycodeschool
 * [Data Structures](https://stepik.org/course/579/syllabus) - Niema Moshiri, Liz Izhikevich (Stepik)
+* [Data Structures](https://youtube.com/playlist?list=PLBlnK6fEyqRj9lld8sWIUNwlKfdUoPd1Y) - Neso Academy (YouTube)
 * [Data Structures and Algorithms](https://youtube.com/playlist?list=PLBZBJbE_rGRV8D7XZ08LK6z-4zPoWzu5H) - CS Dojo
 * [Data Structures and Algorithms](https://www.youtube.com/playlist?list=PLdo5W4Nhv31bbKJzrsKfMpo_grxuLl8LU) - Jenny's lectures CS/IT NET&JRF
-* [Data Structures and Algorithms for Beginners](https://www.youtube.com/watch?v=BBpAmxU_NQo) - Moshfegh Hamedani (Programming with Mosh)
+* [Data Structures and Algorithms for Beginners](https://www.youtube.com/watch?v=BBpAmxU_NQo) - Mosh Hamedani (Programming with Mosh)
+* [Data Structures and Algorithms Full Course](https://youtu.be/LcKqYoX8uH4) - Simplilearn (YouTube, Video length 7:13:56)
```
Comment (davorpa) [inline courses/free-courses-en.md:125]: Expand YT shorten urls

```suggestion
* [Data Structures and Algorithms Full Course](https://www.youtube.com/watch?v=LcKqYoX8uH4) - Simplilearn (YouTube, Video length 7:13:56)
```

**Review by davorpa — CHANGES_REQUESTED** [review]

**Review by davorpa — APPROVED** [review]

**Pull request merged**